model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProvider.java

```diff
@@ -698,7 +637,7 @@ private MapUserSessionEntity createUserSessionEntityInstance(String id, String r
     private MapAuthenticatedClientSessionEntity createAuthenticatedClientSessionEntityInstance(String id, String userSessionId, String realmId,
                                                                                                String clientId, boolean offline) {
         MapAuthenticatedClientSessionEntityImpl clientSessionEntity = new MapAuthenticatedClientSessionEntityImpl();
-        clientSessionEntity.setId(id);
+        clientSessionEntity.setId(id == null ? KeycloakModelUtils.generateId() : id);
```

Comment:
Yes. Otherwise `clientSession`s have no ids because after removing the `clientSessionTx` nothing is assigning ids to it. However, this is true for all nested entities.

Reply:
makes sense, thanks